### ISSUE TITLE
One coafile for one language

### ISSUE THREAD

**Issue by SanketDG** [coala-analyzer/coala#1898]
Currently the design of LanguageDefinition and DocstyleDefintion is such that separate coafiles have to be used for one language. It is okay if this is supported but unified coafiles also need to be supported.

**Comment by SanketDG** [comment]
This issue is different from #1892.

This deals with the design of one coafile for every language, which will includes its keywords, doc-markers etc. For every kind of language/doc parsing currently done or will be done, only the relevant fields should be parsed.

**Comment by abhsag24** [comment]
@SanketDG you mean coalang right? 
so we don't want users to create coalang files right? we want to create coalang files so that users can use them, so in this case having seperate files for seperate things makes sense?  why do we need coalang files  to be united? any use case?

**Comment by SanketDG** [comment]
@abhsag24 Yes coalang :D
Why wouldn't we want users to create coalang files? Yes, we could do the majority of them, but we can't possibly cover every language.

coalang file should be like a specification that users should conform to. 

> so in this case having seperate files for seperate things makes sense?

Yes this does makes sense. I am not sure what I want here, I will have to be do a 1-on-1 with @Makman2 to understand his thoughts about this :D

**Comment by abhsag24** [comment]
@SanketDG  so IIUC what you want is a coalang file that can be used by both `DocStyleDefinition` and `LanguageDefinition` and can be created on the fly by the user, is that correct? 
so we could have those coalang files separated like they always were and have a third type of unified coalang file? , this would in anycase close #1892 right? since that is just the way coalang files  for DocStyleDefinition were defined right?

**Comment by Makman2** [comment]
@abhsag24 this would somehow defeat the purpose of this unifying-idea :P This only adds unnecessary overhead into this "third-type"-coalang-file and our code infrastructure^^

@SanketDG
> so in this case having seperate files for seperate things makes sense?

> Yes this does makes sense.

Where's the problem then? :)

In fact I like the principle of the idea, a language has a set of characteristics/settings that is only language-bound.

**Comment by abhsag24** [comment]
@Makman2 @SanketDG  coalang files are being used more an more now, we need to design a way to merge these two types of coalangs together, soon.

**Comment by Makman2** [comment]
I totally agree, though I think it's not that urgent, as it's just about centralizing those language specific data.

In fact we just need to implement a nice and beautiful key-value store for our language-data purpose^^
